server/src/test/java/com/linecorp/centraldogma/server/internal/httpapi/ContentServiceV1Test.java

```diff
@@ -0,0 +1,369 @@
+        addFooFile();
+        addBarFile();
+
+        // get the list of all files
+        final AggregatedHttpMessage res1 = httpClient.get(CONTENTS_PREFIX + "/**").aggregate().join();
+        final String expectedJson1 =
+                '[' +
+                "   {" +
+                "       \"path\": \"/a/bar.txt\"," +
+                "       \"type\": \"TEXT\"," +
+                "       \"content\" : \"text in the file.\"" +
+                //"   \"pushedAt\": \"${json-unit.ignore}\"" +
+                "   }," +
+                "   {" +
+                "       \"path\": \"/foo.json\"," +
+                "       \"type\": \"JSON\"," +
+                "       \"content\" : {\"a\":\"bar\"}" +
+                //"   \"pushedAt\": \"${json-unit.ignore}\"" +
+                "   }" +
+                ']';
+        final String actualJson1 = res1.content().toStringUtf8();
+        assertThatJson(actualJson1).isEqualTo(expectedJson1);
+
+        final AggregatedHttpMessage res2 = httpClient.get(CONTENTS_PREFIX + "/**?revision=2")
+                                                     .aggregate().join();
+        final String expectedJson2 =
+                '[' +
+                "   {" +
+                "       \"path\": \"/foo.json\"," +
+                "       \"type\": \"JSON\"," +
+                "       \"content\" : {\"a\":\"bar\"}" +
+                //"   \"pushedAt\": \"${json-unit.ignore}\"" +
+                "   }" +
+                ']';
+        final String actualJson2 = res2.content().toStringUtf8();
+        assertThatJson(actualJson2).isEqualTo(expectedJson2);
+    }
+
+    @Test
+    public void deleteFile() throws IOException {
+        addFooFile();
+        addBarFile();
+
+        final String body =
+                '{' +
+                "   \"commitMessage\" : {" +
+                "       \"summary\" : \"Delete foo.json\"" +
+                "   }" +
+                '}';
+        final AggregatedHttpMessage res1 = httpClient
+                .execute(HttpHeaders.of(HttpMethod.DELETE, CONTENTS_PREFIX + "/foo.json"),
+                         HttpData.ofUtf8(body)).aggregate().join();
+        assertThat(res1.headers().status()).isEqualTo(HttpStatus.NO_CONTENT);
+
+        final AggregatedHttpMessage res2 = httpClient.get(CONTENTS_PREFIX + "/**").aggregate().join();
+        // only /a/bar.txt file is left
+        assertThat(Jackson.readTree(res2.content().toStringUtf8()).size()).isOne();
+    }
+
+    @Test
+    public void deleteFileInvalidRevision() throws IOException {
+        addFooFile();
+        addBarFile();
+        final String body =
+                '{' +
+                "   \"commitMessage\" : {" +
+                "       \"summary\" : \"Delete foo.json\"" +
+                "   }" +
+                '}';
+        final AggregatedHttpMessage res = httpClient
+                .execute(HttpHeaders.of(HttpMethod.DELETE, CONTENTS_PREFIX + "/foo.json?revision=2"),
```

Comment:
Do we need convenience methods for this on upstream? like `HttpClient.delete(uri, string body)`? It is not usual cases though.